src/cpl/nuopc/rof_comp_nuopc.F90

```diff
@@ -0,0 +1,886 @@
+       nsrest = nsrBranch
+    else
+       call shr_sys_abort( subname//' ERROR: unknown starttype' )
+    end if
+
+    call RtmVarSet(&
+         caseid_in=caseid, &
+         ctitle_in=ctitle,   &
+         brnch_retain_casename_in=brnch_retain_casename, &
+         nsrest_in=nsrest, &
+         version_in=model_version,     &
+         hostname_in=hostname, &
+         username_in=username)
+
+    call Rtmini()
+
+    ! Initialize memory for input state
+
+    if (rtm_active) then
+       allocate (totrunin(runoff%begr:runoff%endr,nt_rtm))
+    end if
+
+    !--------------------------------
+    ! generate the mesh and realize fields
+    !--------------------------------
+
+    ! determine global index array
+    lsize = runoff%endr - runoff%begr + 1
+    allocate(gindex(lsize))
+    ni = 0
+    do n = runoff%begr,runoff%endr
+       ni = ni + 1
+       gindex(ni) = runoff%gindex(n)
+    end do
+
+    ! create distGrid from global index array
+    DistGrid = ESMF_DistGridCreate(arbSeqIndexList=gindex, rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    deallocate(gindex)
+
+    ! read in the mesh
+    call NUOPC_CompAttributeGet(gcomp, name='mesh_rof', value=cvalue, rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+    EMesh = ESMF_MeshCreate(filename=trim(cvalue), fileformat=ESMF_FILEFORMAT_ESMFMESH, &
+          elementDistgrid=Distgrid, rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    if (masterproc) then
+       write(iulog,*)'mesh file for domain is ',trim(cvalue)
+    end if
+
+    !--------------------------------
+    ! realize actively coupled fields
+    !--------------------------------
+
+    call realize_fields(gcomp,  Emesh, flds_scalar_name, flds_scalar_num, rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+    !--------------------------------
+    ! Create RTM export state
+    !--------------------------------
+
+    call export_fields(gcomp, rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+    ! Set global grid size scalars in export state
+    call State_SetScalar(dble(rtmlon), flds_scalar_index_nx, exportState, &
+         flds_scalar_name, flds_scalar_num, rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+    call State_SetScalar(dble(rtmlat), flds_scalar_index_ny, exportState, &
+         flds_scalar_name, flds_scalar_num, rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+    !----------------------------------------------------------------------------
+    ! Reset shr logging
+    !----------------------------------------------------------------------------
+
+    call shr_file_setLogUnit (shrlogunit)
+
+    call ESMF_LogWrite(subname//' done', ESMF_LOGMSG_INFO)
+
+    !--------------------------------
+    ! diagnostics
+    !--------------------------------
+
+    if (debug > 1) then
+       call State_diagnose(exportState,subname//':ES',rc=rc)
+       if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    endif
+
+#ifdef USE_ESMF_METADATA
```

Comment:
So this is something that CIME will set in the build (USE_ESMF_METADATA)? If not, how do you  turn this on?

Reply:
This should be removed. It is not being used and is no longer relevant. I have removed this and pushed things back to my PR branch.